CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -70,9 +72,7 @@ class CardClient internal constructor(
         try {
             cardAPI.fetchCachedOrRemoteClientID()
         } catch (e: PayPalSDKError) {
-            approveOrderListener?.onApproveOrderFailure(
-                APIClientError.clientIDNotFoundError(e.code, e.correlationID)
-            )
+            notifyApproveOrderFailure(APIClientError.clientIDNotFoundError(e.code, e.correlationID))
```

Comment:
Let the error be handled by the `coroutineExcpetionHandler`:
```suggestion
            throw APIClientError.clientIDNotFoundError(e.code, e.correlationID)
```
and remove the `return` below it

Reply:
@jcnoriega - Hm, making this change results in a `SwallowedException` from Detekt.

Reply:
I reverted this change. I think it boils down to Detekt keeps complaining about a SwallowedException due to the way that CardAPI just wraps the API.fetchCachedOrRemoteClientID() method, which might warrant a larger refactor. Happy to address this in a follow-up PR, though.